src/error.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2019 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::ffi;
+use std::{ffi::CStr, os::raw::c_char};
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Extraction error: '{0}'")]
+    ExtractionError(String),
+
+    #[error("Nix error: '{0}'")]
+    Nix(#[from] nix::Error),
+
+    #[error("Io error: '{0}'")]
+    Io(#[from] std::io::Error),
+
+    #[error("Utf error: '{0}'")]
+    Utf(#[from] std::str::Utf8Error),
+
+    #[error("Try from int error: '{0}'")]
+    TryInt(#[from] std::num::TryFromIntError),
+
+    #[error("Error to create the archive struct, is null")]
+    ArchiveNull,
+
+    #[error("The entry is null, failed to set the pathname")]
+    EntryNull,
+
+    #[error("File not found")]
+    FileNotFound,
+}
+
+pub(crate) fn handle_error(value: i32, archive: *mut ffi::archive) -> Result<()> {
+    if value != ffi::ARCHIVE_OK {
+        return Err(unsafe { Error::from(ffi::archive_error_string(archive)) });
+    }
+
+    Ok(())
+}
+
+impl From<*const c_char> for Error {
+    fn from(out: *const c_char) -> Self {
+        let out = &out;
+        Error::ExtractionError(unsafe { CStr::from_ptr(*out).to_str().unwrap().to_string() })
+    }
```

Comment:
This is actually needed? Couldn't we handle it all using the above?

See: https://github.com/idiv-biodiversity/libarchive-rs/blob/master/libarchive/src/error.rs#L19-L35